src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import duke.task.Task;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.util.Date;
+import java.time.format.DateTimeFormatter;
+import java.text.SimpleDateFormat;
+
+public class Parser {
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+    private static final String DELETE = "delete";
+    private static final String DONE = "done";
+    private static final String LIST = "list";
+    private static final String BYE = "bye";
+    private static final String FIND = "find";
+    public static final String FAIL = "fail";
+
+    /**
+     * Parse user input command and deal with incorrect input.
+     *
+     * @param inputLine User input command line.
+     * @return The parsed command line.
+     * @throws DukeException Throw invalid or incomplete exceptions.
+     */
+    public static String getCommand(String inputLine) throws DukeException{
+        try{
+            if (inputLine.equals(LIST)) {
```

Comment:
Great that you are using constants for this :+1: . As you have many else if's, maybe switch shouldn't be more appropriate?